beacon-chain/execution/engine_client.go

```diff
@@ -94,14 +95,24 @@ func (s *Service) NewPayload(ctx context.Context, payload interfaces.ExecutionDa
 	ctx, cancel := context.WithDeadline(ctx, d)
 	defer cancel()
 	result := &pb.PayloadStatus{}
-	payloadPb, ok := payload.Proto().(*pb.ExecutionPayload)
-	if !ok {
+
+	proto := payload.Proto()
+	payloadOld, okOld := proto.(*pb.ExecutionPayload)
+	payloadNew, okNew := proto.(*pb.ExecutionPayload4844)
```

Comment:
It's ofc more efficient to not parse them both. Do one, then the other.

Reply:
might suggest during a switch statement on the type, rather than casting twice

Reply:
Switching on the type of `payload` doesn't work because it matches neither *pb.ExecutionPayload nor *pb.ExecutionPayload4844. Those generated structs are wrapped by `WrappedExecutionPayload` to make them match the ExecutionData interface. 

But they do cast okay, I guess.

Reply:
another thing we could do is add a `getRPCExecutionPayload()` to the `ExecutionPayload` interface that returns the `executionPayloadJSON`. There already exists an `executionPayloadJSON` struct used in json_marshal_unmarshal.go for the JSON codec. This lets us avoid explicit type switching, and rely on the type system to ensure we're doing the right thing.